pkg/profile/helm.go

```diff
@@ -0,0 +1,112 @@
+package profile
+
+import (
+	"reflect"
+	"strings"
+
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HelmReleaseRequiresUpdate checks if the helm release resource requires updating
+func HelmReleaseRequiresUpdate(existingRes, newRes *helmv2.HelmRelease) bool {
+	switch {
+	case existingRes.Spec.Chart.Spec.Chart != newRes.Spec.Chart.Spec.Chart:
+		return true
+	case existingRes.Spec.Chart.Spec.Version != newRes.Spec.Chart.Spec.Version:
+		return true
+	case !reflect.DeepEqual(existingRes.Spec.Chart.Spec.SourceRef, newRes.Spec.Chart.Spec.SourceRef):
+		return true
+	case !reflect.DeepEqual(existingRes.Spec.Values, newRes.Spec.Values):
+		return true
+	case !reflect.DeepEqual(existingRes.Spec.ValuesFrom, newRes.Spec.ValuesFrom):
+		return true
+	default:
+		return false
+	}
+}
+
+// HelmRepoRequiresUpdate checks if the helm repository resource requires updating
+func HelmRepoRequiresUpdate(existingRes, newRes *sourcev1.HelmRepository) bool {
+	switch {
+	case existingRes.Spec.URL != newRes.Spec.URL:
+		return true
+	default:
+		return false
+	}
+}
+
+func (p *Profile) makeHelmRepository(url string, name string) (*sourcev1.HelmRepository, error) {
+	helmRepo := &sourcev1.HelmRepository{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      p.makeHelmRepoName(name),
+			Namespace: p.subscription.ObjectMeta.Namespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       sourcev1.HelmRepositoryKind,
+			APIVersion: sourcev1.GroupVersion.String(),
+		},
+		Spec: sourcev1.HelmRepositorySpec{
+			URL: url,
+		},
+	}
+	return helmRepo, nil
+}
+
+func (p *Profile) makeHelmRepoName(name string) string {
+	repoParts := strings.Split(p.subscription.Spec.ProfileURL, "/")
+	repoName := repoParts[len(repoParts)-1]
+	return join(p.subscription.Name, repoName, p.subscription.Spec.Branch, name)
+}
+
+func (p *Profile) makeHelmRelease(artifact profilesv1.Artifact) (*helmv2.HelmRelease, error) {
```

Comment:
This never returns an error?

Reply:
this was already here, just moved it. Will update